Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +919,99 @@ export class OtherSettings {
+    private getAsBoolean(sectionName: string, settingName: string, resource: any, defaultValue: boolean): boolean {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = section.get<any>(settingName);
+        if (isBoolean(value)) {
+            return value;
+        }
+        const setting = section.inspect<any>(settingName);
+        if (setting?.defaultValue === undefined || setting.defaultValue === null) {
+            this.logValidationError(sectionName, settingName, "no default value");
+            return defaultValue;
+        }
+        return setting.defaultValue;
+    }
+
+    private getAsNumber(sectionName: string, settingName: string, resource: any, defaultValue: number, minimum?: number, maximum?: number): number {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = section.get<any>(settingName);
+        // Validates the value is a number and clamps it to the specified range. Allows for undefined maximum or minimum values.
+        if (isNumber(value)) {
+            if (minimum !== undefined && value < minimum) {
+                return minimum;
+            }
+            if (maximum !== undefined && value > maximum) {
+                return maximum;
+            }
+            return value;
+        }
+        const setting = section.inspect<any>(settingName);
+        if (setting?.defaultValue === undefined || setting.defaultValue === null) {
+            this.logValidationError(sectionName, settingName, "no default value");
+            return defaultValue;
+        }
+        return setting.defaultValue;
+    }
+
+    private getAsAssociations(sectionName: string, settingName: string, resource?: any): Associations {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = section.get<any>(settingName);
+        if (isValidMapping(value, isString, isString)) {
+            return value as Associations;
+        }
+        const setting = section.inspect<any>(settingName);
+        if (setting?.defaultValue === undefined || setting.defaultValue === null) {
+            this.logValidationError(sectionName, settingName, "no default value");
+        }
+        return setting?.defaultValue as Associations;
+    }
+
+    private getAsExcludes(sectionName: string, settingName: string, resource?: any): Excludes {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = section.get<any>(settingName);
+        if (isValidMapping(value, isString, (val) => isBoolean(val) || isValidWhenObject(val))) {
+            return value as Excludes;
+        }
+        const setting = section.inspect<any>(settingName);
```

Comment:
This `setting` variable doesn't have a type -- can you add a type?

Same with "section" and "value" variables.

Reply:
I don't think the types add any value here. `value` is `any`, `setting` is a crazy long untyped object, `section` is `vscode.WorkspaceConfiguration`